### ISSUE TITLE
TS files with teletext sub refuses to play

### ISSUE THREAD

**Issue by atrus05** [plexinc/plex-media-player#516]
I have a .ts file with two subs embedded. One is a DVB_TELETEXT sub. If I choose that one the episode does not even start. PMP throws a black screen saying
```There was a problem playing this item.```
```no audio or video data played```

That is of course annoying, but even more annoying is that if I happen to choose that sub, there is no way for me to change the sub back, as we do not have the ability to change the sub before start of playback in PMP still.

Here is a sample file of that ts-file, and I have double checked the sample and it exhibits the same problem as the full ts-file: https://www.dropbox.com/s/sud84ii9h157493/tut-sub-issue.ts?dl=1

Here is the part of the log from when I hit "play":

**Comment by wm4** [comment]
Yeah, there's no `dvb_teletext` decoder. So attempting to play it will fail. You should be able to make it work by unselecting it. You can also try `stop-playback-on-init-failure=no` in mpv.conf.

**Comment by atrus05** [comment]
That is a nifty little mpv command, thanks. However, going forward I recommend we perhaps handle it a bit better. For example I suggest:
1. Give a correct error message.
2. If the init-failure gateway will remain, perhaps not include the lack of a teletext-decoder as a init-failure in that sense.

**Comment by wm4** [comment]
> Give a correct error message.

Strictly speaking the error message is correct, but yeah, point taken.

> If the init-failure gateway will remain, perhaps not include the lack of a teletext-decoder as a init-failure in that sense.

It'd be weird if we played only audio or video if the other fails. That's what it's mainly for. Subtitles are just another failure case that is affected by this. Not sure if adding an exception for subtitles is reasonable.

Maybe web-client should not try to select those subtitle tracks by default though.

**Comment by LongChair** [comment]
@atrus05 : i see associated web issue being closed. Can we close this one ?

**Comment by mseeley** [comment]
PMP Desktop and TV mode are able to change selected streams from the preplay now; 😅. Neither layout is capable of playing the dvb_teletext stream:

![screen shot 2018-11-26 at 2 35 49 pm](https://user-images.githubusercontent.com/30019/49050898-a7a19c00-f189-11e8-9b5c-2572e9b0e21c.png)

![screen shot 2018-11-26 at 2 34 42 pm](https://user-images.githubusercontent.com/30019/49050900-aa03f600-f189-11e8-9dda-75057fada684.png)

Switching to the dvb_subtitle stream does result in subtitles displaying.

An improved error message and the ability to change streams from preplay helps users work around this incompatibility. I'll leave this open at least until I understand better about our handling of this stream format.

**Comment by rcombs** [comment]
This likely just requires us to build against the libzvbi library and enable the dvb_teletext decoder.

**Comment by mseeley** [comment]
Nice, that's good news. Thank you @rcombs. 😄 

I'll tag this for addressing when updating MPV.
